R/loadtest.R

```diff
@@ -60,6 +60,29 @@ parse_url <- function(url){
   list(protocol = protocol, domain = domain, path = path, port = port)
 }
 
+#' Encode HTML/XML entities
+#'
+#' Jemeter's documentation in XML which means special characters like &, <, >
+#' need to be encoded as HTML entities. This function does that without dependencies.
+#'
+#' @param stringified_body the request body as a single string
+#'
+#' @return a string
+#'
+#' @examples
+#' encode_html_entities('{"title":"this & that"}')
+encode_html_entities <- function(stringified_body) {
+  body_noamp <- gsub("&", "&amp;", stringified_body)
+  body_noquot <- gsub("\"", "&quot;", body_noamp)
+  body_nogt <- gsub(">", "&gt;", body_noquot)
+  body_nolt <- gsub("<", "&lt;", body_nogt)
+  body_noapos <- gsub("\'", "&apos;", body_nolt)
```

Comment:
I'm not a fan of overwriting variables e.g. `x <- x + 1` but maybe here it's ok. Let me know what you think and I'll change accordingly.